orchestrator/nodeManager.js

```diff
@@ -50,229 +58,323 @@ class NodeManager {
     }
 
     logger.debug('Node manager is using the ' + engine + ' engine');
+  }
 
+  /**
+   * Creates a mongodb collection for persinting remote node configurations (promise).
+   *
+   * @param {*} tenant identifier of the tenant.
+   */
+  _createMongoDBCollection(tenant) {
+    return MongoManager.get().then((client) => {
+      let collection = client.db(`flowbroker_${tenant}`).collection('remoteNode');
+      return collection;
+    }).catch(() => {
+      thrown(new Error (`Failed to create mongoDB collection for tenant ${tenant}.`));
+    });
   }
 
-  startContainer(tenant) {
-    this.collection[tenant].find().toArray()
+  /**
+   * Loads remote nodes for the given tenant.
+   *
+   * @param {*} tenant identifier of the tenant.
+   */
+  _loadRemoteNodes(tenant) {
```

Comment:
I think in this function the `await` could be used. Therefore it would remove any `then` and its nesting nature.

Reply:
Ok. I changed to remove most of the promise chains.
See #135